tree/treeplayer/src/TSimpleAnalysis.cxx

```diff
@@ -0,0 +1,384 @@
+ *                                                                       *
+ * For the licensing terms see $ROOTSYS/LICENSE.                         *
+ * For the list of contributors see $ROOTSYS/README/CREDITS.             *
+ *************************************************************************/
+
+#include "TSimpleAnalysis.h"
+
+#include "TFile.h"
+#include "TChain.h"
+#include "TH1.h"
+#include "TError.h"
+#include "TKey.h"
+
+#include <string>
+#include <fstream>
+#include <vector>
+#include <map>
+#include <iostream>
+
+/** \class TSimpleAnalysis
+A TSimpleAnalysis object allow to, given an input file, create a
+.root file in which are saved histograms that the user wants to
+create. The configuration file needs to set the parameters that allow
+to create the histograms. This file has a fixed sintax, in which are
+allowed comments (marked with the '#' sign) at the beginning of the
+line (less than whitespaces), or at the end of the configuration
+line. Empty or comment lines are skipped from the configuration procedure.
+Here an example of configuration file:
+```
+#This is an example of configuration file
+file_output.root   #the output file in which are stored histograms
+
+#The next line has the name of the tree of the input data. It is
+#optional if there is exactly one tree in the first input file
+ntuple   #name of the input tree
+
+#The lines of the next block correspond to .root input files that
+#contain the tree
+hsimple1.root   #first .root input file
+hsimple2.root   #second .root input file
+
+#The next block is composed by lines that allow to configure the
+#histograms. They have to be the follow syntax:
+#name = expression if cut
+#which corresponds to chain->Draw("expression >> name", "cut")
+#i.e. it will create a histogram called name and store it in
+#file_output.root.
+#"if cut" is optional
+hpx=px if px<-3   #first histogram
+hpxpy=px:py    #second histogram
+
+#End of the configuration file
+```
+*/
+
+////////////////////////////////////////////////////////////////////////////////
+/// Delete leading and trailing white spaces in a string.
+///
+/// param[in] line line read from the input file
+
+static void DeleteSpaces(std::string& line)
+{
+   // Delete leading spaces
+   std::size_t firstNotSpace = line.find_first_not_of(" \t");
+   if (firstNotSpace != std::string::npos)
+      line = line.substr(firstNotSpace, line.size() - firstNotSpace);
+   else
+      line.clear();
+   // Delete trailing spaces
+   std::size_t lastNotSpace = line.find_last_not_of(" \t");
+   if (lastNotSpace != std::string::npos)
+      line = line.substr(0, lastNotSpace + 1);
+}
+
+////////////////////////////////////////////////////////////////////////////////
+/// Handle the expression lines of the input file in order to pass the
+/// elements to the members of the object.
+///
+/// param[in] line line or read from the input file either the expression passed to the constructor
```

Comment:
`line - histogram expression, either read from the config file or passed as expression to the constructor.`